compiler/rustc_resolve/src/access_levels.rs

```diff
@@ -214,13 +213,8 @@ impl<'r, 'ast> Visitor<'ast> for AccessLevelsVisitor<'ast, 'r> {
                     }
                 }
             }
-            ast::ItemKind::Trait(ref trait_kind) => {
-                for nested in trait_kind.items.iter() {
```

Comment:
Traits and enums are also "modules" from name resolution point of view.
So you can use `self.set_exports_access_level(self.r.local_def_id(item.id));` in `ItemKind::Enum`s and `ItemKind::Trait`s, like in `ItemKind::Mod`.

Without that items inside traits are not visited at all if this code is removed.